checker/src/main/java/org/checkerframework/checker/mustcall/MustCallTransfer.java

```diff
@@ -0,0 +1,375 @@
+   * implicit string conversion.
+   */
+  private @MonotonicNonNull AnnotationMirror defaultStringType;
+
+  /**
+   * Create a MustCallTransfer.
+   *
+   * @param analysis the analysis
+   */
+  public MustCallTransfer(CFAnalysis analysis) {
+    super(analysis);
+    atypeFactory = (MustCallAnnotatedTypeFactory) analysis.getTypeFactory();
+    ProcessingEnvironment env = atypeFactory.getChecker().getProcessingEnvironment();
+    treeBuilder = new TreeBuilder(env);
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitStringConversion(
+      StringConversionNode n, TransferInput<CFValue, CFStore> p) {
+    // Implicit String conversions should assume that the String's type is
+    // whatever the default for String is, not that the conversion is polymorphic.
+    TransferResult<CFValue, CFStore> result = super.visitStringConversion(n, p);
+    AnnotationMirror defaultStringType =
+        this.defaultStringType != null
+            ? this.defaultStringType
+            : atypeFactory
+                .getAnnotatedType(TypesUtils.getTypeElement(n.getType()))
+                .getAnnotationInHierarchy(atypeFactory.TOP);
+    LocalVariableNode temp = getOrCreateTempVar(n);
+    if (temp != null) {
+      JavaExpression localExp = JavaExpression.fromNode(temp);
+      insertIntoStores(result, localExp, defaultStringType);
+    }
+    return result;
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitMethodInvocation(
+      MethodInvocationNode n, TransferInput<CFValue, CFStore> in) {
+    TransferResult<CFValue, CFStore> result = super.visitMethodInvocation(n, in);
+
+    updateStoreWithTempVar(result, n);
+    if (!atypeFactory.getChecker().hasOption(MustCallChecker.NO_ACCUMULATION_FRAMES)) {
+      List<JavaExpression> targetExprs =
+          getCreatesObligationExpressions(n, atypeFactory, atypeFactory);
+      for (JavaExpression targetExpr : targetExprs) {
+        AnnotationMirror defaultType =
+            atypeFactory
+                .getAnnotatedType(TypesUtils.getTypeElement(targetExpr.getType()))
+                .getAnnotationInHierarchy(atypeFactory.TOP);
+
+        if (result.containsTwoStores()) {
+          CFStore thenStore = result.getThenStore();
+          lubWithStoreValue(thenStore, targetExpr, defaultType);
+
+          CFStore elseStore = result.getElseStore();
+          lubWithStoreValue(elseStore, targetExpr, defaultType);
+        } else {
+          CFStore store = result.getRegularStore();
+          lubWithStoreValue(store, targetExpr, defaultType);
+        }
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Computes the LUB of the current value in the store for expr, if it exists, and defaultType.
+   * Inserts that LUB into the store as the new value for expr.
+   *
+   * @param store a CFStore
+   * @param expr an expression that might be in the store
+   * @param defaultType the default type of the expression's static type
```

Comment:
Why is this variable named `defaultType`?  I think the method doesn't depend on it being a default.

Reply:
Technically it does not, but the intended use case is that this method will compute the LUB of the current type in the store and the default type. Since the default type is computed elsewhere, this method takes it as a parameter (and the method is therefore more general: you're right that default type could actually be any type). That said, I'd prefer to keep the name to preserve the intent of the method. It's private, so it won't be used elsewhere.

Reply:
If you feel strongly or have a compelling argument for why changing the name to make the method's contract more general is good idea, I'm happy to make the change.